libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -121,11 +141,25 @@ bool HTTPClient::begin(String url, String httpsFingerprint)
 bool HTTPClient::begin(String url)
 {
     _transportTraits.reset(nullptr);
-    _port = 80;
-    if (!beginInternal(url, "http")) {
+    if (!beginInternal(url, nullptr)) {
+        return false;
+    }
+    if (_protocol == "https" || _port == 443) {
+        _port = 443;
+        _transportTraits = TransportTraitsPtr(new TLSTraits());
+    }
+    else {
+        _port = 80;
+        _transportTraits = TransportTraitsPtr(new TransportTraits());
+    }
+    _tcp = _transportTraits->create();
+
+    if(!_tcp->connect(_host.c_str(), _port)) {
```

Comment:
No saving the CA is a bad idea as it would consume a lot of RAM. I've been trying to explicitly avoid storing the CA.